File: C#-Server+Client/BaseLibrary/communication/command/ACommandFactory.cs
using BaseLibrary.communication.protocol;

namespace BaseLibrary.communication.command {
    /// <summary>
    /// Factory for command
    /// </summary>
    public abstract class ACommandFactory : AFactory<ACommand.Sendable, ACommand> {
        /// <summary>
        /// Try to Transferable and if it is possible store serialization of transfer object like string.
        /// </summary>
        /// <param name="c"></param>
        /// <returns></returns>
        public override bool IsSerializable(ACommand c) {
            if (IsTransferable(c)) {
                cacheForSerialize.Cached(c, Transfer(c).Serialize());
                return true;
            }
            return false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/ACommandFactory.java
package BaseLibrary.communication.command;

import BaseLibrary.communication.protocol.AProtocol;
import BaseLibrary.communication.protocol.IFactory.AFactory;

/**
    * Factory for command
    */
    public abstract class ACommandFactory extends AFactory<ACommand, ACommand> {
        /**
        * Try to Transferable and if it is possible store serialization of transfer object like String.
        */
    	@Override
    	public boolean isSerializeable(ACommand c, AProtocol protocol) {
            if (isTransferable(c)) {
                cacheForSerialize.cached(c, transfer(c).serialize(protocol));
                return true;
            }
            return false;
        }
}
